pkg/providers/tinkerbell/validator.go

```diff
@@ -132,7 +134,28 @@ func (v *Validator) ValidateHardwareConfig(ctx context.Context, hardwareConfigFi
 		return fmt.Errorf("failed validating Secrets in hardware config: %v", err)
 	}
 
-	logger.MarkPass("Hardware Config is valid")
+	if err := v.ValidateBMCSecretCreds(ctx, v.hardwareConfig); err != nil {
```

Comment:
I feel this can be its own validation, inside `SetupAndValidateCreateCluster`. Yes, we are using information from hardwareConfig but we are also doing API calls and connectivity checks so I think it can be moved out of this function.